daemon/config/config.go

```diff
@@ -623,6 +636,24 @@ func ValidateMinAPIVersion(ver string) error {
 	return nil
 }
 
+func validateFeatures(features map[string]bool) error {
+	if len(features) > MaxFeatures {
+		return errors.Errorf("too many features – expected max %d, found %d", MaxFeatures, len(features))
+	}
+	for k := range features {
+		if strings.Contains(k, "=") {
+			return errors.Errorf("invalid feature – key cannot contain '=': %s", k)
+		}
+		if strings.Contains(k, ",") {
+			return errors.Errorf("invalid feature – key cannot contain ',': %s", k)
+		}
+		if len(k) > MaxFeatureKeyLen {
+			return errors.Errorf("feature name length cannot be over %d: %s", MaxFeatureKeyLen, k)
+		}
+	}
```

Comment:
IMO, instead of blacklisting specific characters, I'd just upfront restrict it to alphanumeric characters, `-` and `.`.
We can always extend the whitelist in future, where it's harder to backoff from character that was previously valid.

Reply:
Up until now, we haven't had any validation! But I agree for sure.